src/lib/y2network/network_manager/connection_config_writer.rb

```diff
@@ -31,16 +31,24 @@ class ConnectionConfigWriter
       def write(conn, old_conn = nil)
         return if conn == old_conn
 
-        file = CFA::NmConnection.new(SYSTEM_CONNECTIONS_PATH.join(conn.name).sub_ext(FILE_EXT))
+        path = SYSTEM_CONNECTIONS_PATH.join(conn.name).sub_ext(FILE_EXT)
+        file = CFA::NmConnection.new(path)
         handler_class = find_handler_class(conn.type)
         return nil if handler_class.nil?
 
+        ensure_permissions(path) unless ::File.exist?(path)
```

Comment:
You could also pass the mode to the `CFA::NmConnection` constructor if you prefer or even in the `save` method. Anyway, it is OK here.

Reply:
Yep, was not sure about the best place by now, so did it explicitly here by now.